### PR TITLE
Some updates to release process docs

### PR DESCRIPTION
**Pull request by esethna**
@amyblais can I have your help reviewing this PR and helping make any tweaks you feel are necessary to fully explain the tick-tick release cycle?

I renamed the FAQ page to "Release Overview" so we didn't break links (Added the FAQ to the bottom of the new doc)

I think we should expand on some of the FAQs in this doc if you can help with that? I'm finding them hard to follow/read. I think it would be valuable to explain the branching process in a bit more detail.

### CODE REVIEW

**Comment by amyblais** [comment]
Tick-tock section looks good.

Agree some of the FAQ sections could be expanded on - would you say questions 8, 9, 17 and 19 are the main ones that need expanding, or did you see other ones that could have more information?

**Comment by hanzei** [comment]
@esethna Would you please take a look at the above question?

**Comment by esethna** [comment]
Ya but 0/5, I'll let you own this document - I think the goal is to have a place to point people to understand exactly how the release process and branching works. May need to add more questions or update them,

For example, Q9 does not explain community daily, or the timeline of when the servers are updated

**Comment by esethna** [comment]
Happy to review again when you think it's in a good state

File: source/process/release-faq.rst [hunk 1]
```diff
@@ -1,74 +1,128 @@
...
+**Q: How does quality release work?**
 
-**Q7: How to remove feature/bug from a release?**
+- A: Bugs are branched off from the previous feature release.
  
- - A: Revert from release branch. Optionally revert from master.
+**Q: How is cherry-picking done?**
 
-**Q8: How are NOTICE.txt PRs submitted?**
+- A:
```
Comment (amyblais) [inline source/process/release-faq.rst]: @cpanato Can you help with more details on this? Is this documented somewhere?

File: source/process/release-faq.rst [hunk 2]
```diff
@@ -1,74 +1,128 @@
...
  
- - A: Revert from release branch. Optionally revert from master.
+**Q: How is cherry-picking done?**
 
-**Q8: How are NOTICE.txt PRs submitted?**
+- A: 
 
- - A: Same as Q3.
+**Q: What is community.mattermost kept on?**
+
```
Comment (amyblais) [inline source/process/release-faq.rst]: @cpanato Can you help with more details on this?

File: source/process/release-faq.rst [hunk 3]
```diff
@@ -1,74 +1,128 @@
...
 
-**Q8: How are NOTICE.txt PRs submitted?**
+- A: 
 
- - A: Same as Q3.
+**Q: What is community.mattermost kept on?**
+ 
+- A: For Feature releases community.mattermost is kept on the RCs. For Bug Fix releases keep it on master.
+ 
+**Q: What is community-daily.mattermost kept on?**
```
Comment (amyblais) [inline source/process/release-faq.rst]: @cpanato Can you help with more details on this?

File: source/process/release-faq.rst [hunk 4]
```diff
@@ -1,74 +1,128 @@
...
+- A: PR with changes was merged `here <https://github.com/mattermost/mattermost-developer-documentation/pull/182>`__.
 
-**Q13: What changes were made to the team release process to account for the rotating feature and bug fix releases? https://docs.mattermost.com/process/release-process.html**
+**Q: What changes were made to the team release process to account for the rotating feature and quality releases? https://docs.mattermost.com/process/release-process.html**
 
- - A: Separate checklists for `Bug Fix release <https://docs.mattermost.com/process/bug-fix-release.html>`__ and `Feature release <https://docs.mattermost.com/process/feature-release.html>`__ were created.
+- A: Separate checklists for `Bug Fix release <https://docs.mattermost.com/process/bug-fix-release.html>`__ and `Feature release <https://docs.mattermost.com/process/feature-release.html>`__ were created.
 
-**Q14: How does Redux branching work?**
+**Q: How does Redux branching work?**
```
Comment (amyblais) [inline source/process/release-faq.rst]: @cpanato Can you help with more details on this?

File: source/process/release-faq.rst [hunk 5]
```diff
@@ -1,74 +1,128 @@
...
- - A: For Feature releases community.mattermost is kept on the RCs. For Bug Fix releases keep it on master. 
+- A: PRs are submitted and merged to master (no change).
+
+**Q: How does quality release work?**
 
-**Q7: How to remove feature/bug from a release?**
+- A: Bugs are branched off from the previous feature release.
  
- - A: Revert from release branch. Optionally revert from master.
+**Q: How is cherry-picking done?**
```
Comment (amyblais) [inline source/process/release-faq.rst]: @cpanato Can you help with more details on this? Is this documented somewhere?

**Comment by jasonblais** [comment]
@amyblais are there next steps on this PR?
